api/api_public_blockchain.go

```diff
@@ -407,6 +407,76 @@ func (s *PublicBlockChainAPI) DoEstimateGas(ctx context.Context, b Backend, args
 	return hexutil.Uint64(hi), nil
 }
 
+type EstimateGasTraceResult struct {
+	// Gas is an estimate of the amount of gas needed to execute the given transaction against the latest block.
+	Gas hexutil.Uint64 `json:"gas"`
+	// Trace contains trace result. Empty (undefined) Trace means estimation was successful.
+	Trace interface{} `json:"trace,omitempty"`
+}
+
+// EstimateGasWithTrace returns an estimated gas and trace if it is going to be reverted.
+// NOTE: This method is an unofficial, blockswords fork only method to help debugging.
+func (s *PublicBlockChainAPI) EstimateGasWithTrace(ctx context.Context, args CallArgs) (*EstimateGasTraceResult, error) {
+	b := s.b
+	gasCap := s.b.RPCGasCap()
+	// Following logic is mostly a copy of `DoEstimateGas` with trace enabled on revert.
+	// This is done for the sake of ease of maintenance against upstream.
```

Comment:
Reason for the ctrl-c+v is to have the original method maintain their original own behavior.